clients/spark-extensions/src/test/java/org/projectnessie/spark/extensions/ITNessieStatements.java

```diff
@@ -99,8 +100,50 @@ void testCreateBranchInExists() throws NessieNotFoundException {
   }
 
   @Test
-  void testCreateBranchIn() throws NessieNotFoundException {
+  public void testRefreshAfterMergeWithIcebergTableCaching() {
+    List<Object[]> result = sql("CREATE BRANCH %s IN nessie", refName);
+    assertEquals("created branch", row("Branch", refName, hash), result);
+    assertThat(sql("USE REFERENCE %s IN nessie", refName))
+        .hasSize(1)
+        .containsExactly(row("Branch", refName, hash));
+
+    sql("CREATE TABLE nessie.tbl (id int, name string)");
+    sql("INSERT INTO nessie.tbl select 23, \"test\"");
+    assertThat(sql("SELECT * FROM nessie.tbl")).hasSize(1).containsExactly(row(23, "test"));
+
+    sql(String.format("MERGE BRANCH %s INTO main in nessie", refName));
+    assertThat(sql("SELECT * FROM nessie.`tbl@main`")).hasSize(1).containsExactly(row(23, "test"));
+    assertThat(sql("SELECT * FROM nessie.`tbl@%s`", refName))
+        .hasSize(1)
+        .containsExactly(row(23, "test"));
+    sql("INSERT INTO nessie.tbl select 24, \"test24\"");
+    assertThat(sql("SELECT * FROM nessie.`tbl@main`")).hasSize(1).containsExactly(row(23, "test"));
+    assertThat(sql("SELECT * FROM nessie.tbl"))
+        .hasSize(2)
+        .containsExactlyInAnyOrder(row(23, "test"), row(24, "test24"));
+
+    // this still sees old data because tbl@testBranch is being cached in Iceberg
+    // due to "spark.sql.catalog.nessie.cache-enabled=true" by default
+    assertThat(sql("SELECT * FROM nessie.`tbl@%s`", refName))
+        .hasSize(1)
+        .containsExactly(row(23, "test"));
```

Comment:
Could this be flaky, e.g. if the cache expires for some behind-the-scenes reason?

Reply:
the cache is only invalidated when the table is being renamed or dropped, so shouldn't be flaky here